WrightTools/_dataset.py

```diff
@@ -25,8 +27,53 @@ class Dataset(h5py.Dataset):
     def __getitem__(self, index):
         if not hasattr(index, '__iter__'):
             index = [index]
-        lis = [min(s - 1, i) if not isinstance(i, slice) else i for s, i in zip(self.shape, index)]
-        return super().__getitem__(tuple(lis))
+        index = wt_kit.valid_index(index, self.shape)
+        return super().__getitem__(index)
+
+    def __iadd__(self, value):
+        def f(dataset, s, value):
+            if hasattr(value, 'shape'):
+                dataset[s] += value[wt_kit.valid_index(s, value.shape)]
+            else:
+                dataset[s] += value
+        self.chunkwise(f, value=value)
+        return self
+
+    def __imul__(self, value):
+        def f(dataset, s, value):
+            if hasattr(value, 'shape'):
+                dataset[s] *= value[wt_kit.valid_index(s, value.shape)]
+            else:
+                dataset[s] *= value
+        self.chunkwise(f, value=value)
+        return self
+
+    def __ipow__(self, value):
+        def f(dataset, s, value):
+            if hasattr(value, 'shape'):
+                dataset[s] **= value[wt_kit.valid_index(s, value.shape)]
+            else:
+                dataset[s] **= value
+        self.chunkwise(f, value=value)
+        return self
+
+    def __isub__(self, value):
+        def f(dataset, s, value):
+            if hasattr(value, 'shape'):
+                dataset[s] -= value[wt_kit.valid_index(s, value.shape)]
+            else:
+                dataset[s] -= value
+        self.chunkwise(f, value=value)
+        return self
+
+    def __itruediv__(self, value):
```

Comment:
Should we add `__ifloordiv__` and `__imod__` for completeness sake

Reply:
my vote is to add them later, only if we can think of some good reason for their existance